src/main.cpp

```diff
@@ -450,6 +573,40 @@ class gfxPipelineHandler {
         UpdateWindow(hWND);
         SetFocus(hWND);
     }
+
+    static void Install() {
+        InstallPatch("Enables pointer in windowed mode.", { 0x90, 0x90 }, {
+            0x4F136E,
+        });
+
+        if (!datArgParser::Get("16bit")) {
+            InstallPatch("Creates the initial window in 32-bit mode.", { 32 }, {
+                0x401475,
+                0x401477,
+            });
+        }
+
+        InstallCallback("gfxPipeline::SetRes", "Enables extra resolution parameters (e.g. '-window')",
```

Comment:
Yes, but the game doesn't detect whether or not the current display supports 32-bit. I had to add that check in SetRes, hence the hook calls.

Maybe I can move the check to gfxApplySettings instead, I didn't think of that before.